### DISCUSSION TITLE
capacitor-file-picker is not obeying permissions settings of photo library

### DISCUSSION THREAD

**Discussion by venna-kalyani** [capawesome-team/capacitor-plugins#235] [General]
We came across a scenario and verified, Filepicker.pickMedia() method is not returning photos based on photos permission that the user has selected (ex: limited access - expecting to return only limited selected photos)

So we are looking for alternative and using Camera.pickLimitedLibraryPhotos() method, when user having limited access permission, while using we get the below output 

[{"path":"file:///private/var/mobile/Containers/Data/Application/5A9905CD-7B18-4113-88C7-FC22E13A8ED6/tmp/photo-1.jpg","webPath":"capacitor://localhost:8080/_capacitor_file_/private/var/mobile/Containers/Data/Application/5A9905CD-7B18-4113-88C7-FC22E13A8ED6/tmp/photo-1.jpg","format":”jpeg”}]

We need the actual file name here for further handling. Any way to get that actual file name here.? 

Either way let me know how we can achieve this.

"@capacitor/camera": "^4.1.5",
"@capawesome/capacitor-file-picker": "^0.6.3",

![Image (3) (1)](https://github.com/user-attachments/assets/4e29a148-ebc7-42f9-b781-4739cd817af7)

As we have plan to move to production in couple of weeks we need this for further handling.

Is there any plan to handle this permission(limited selection) based photos selection in future as part of capacitor-file-picker, let us know.

**Comment by robingenz** [comment]
> Is there any plan to handle this permission(limited selection) based photos selection in future as part of capacitor-file-picker, let us know.

For now, this is not planned. As already written via email, the [Capacitor File Picker](https://capawesome.io/plugins/file-picker/) plugin uses the [PHPickerViewController](https://developer.apple.com/documentation/photokit/phpickerviewcontroller), which does not require any permissions and therefore is not affected by this setting (see [here](https://stackoverflow.com/a/76228978/6731412)).
